src/Http/Middleware/Telemetry.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Enjin\Platform\Http\Middleware;
+
+use Closure;
+use Enjin\Platform\Enums\Global\PlatformCache;
+use Enjin\Platform\Services\PhoneHomeService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use Symfony\Component\HttpFoundation\Response;
+use Throwable;
+
+class Telemetry
+{
+    public function __construct(protected PhoneHomeService $service) {}
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  Closure(Request): (Response)  $next
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        if ($this->service->isEnabled()) {
+            try {
+                Cache::lock(PlatformCache::TELEMETRY_CACHE_LOCK->value, 900)->get(fn () => $this->service->phone());
```

Comment:
```suggestion
                if (Cache::lock(PlatformCache::TELEMETRY_CACHE_LOCK->value, 900)->get()) {
                    $this->service->phone();
                }
```

Passing a `Closure` to `get()` will cause the cache lock to be released once the execution of the `Closure` has completed (successfully or not).

```php
if ($result && is_callable($callback)) {
    try {
        return $callback();
    } finally {
        $this->release();
    }
}
```

Therefore, to make this behave correctly, you simply want to check if you were able to acquire the lock or not by calling `get()` without any `$callback`.

This way, if we acquire the lock, then it's the first time in 15 minutes. If not, then we've already called `phone()` in the past 15 minutes and shouldn't do so again.